### DISCUSSION TITLE
The possibility of dolls system(Avatar/Custom Character)?

### DISCUSSION THREAD

**Discussion by allenhsu-17live** [google/filament#4269] [Q&A]
Is there any teaching or any example that can be done like a custom doll system.
It means that I can dynamically load many model components, and these components can be dynamically added or deleted on the character ?

like this : https://www.youtube.com/watch?v=Ch1aJljEYp4
And not only change texture, I think we can add equipment and other models

**Comment by jnmaloney** [comment]
I think samples/gltf_viewer.cpp can do most of this, you can add components dynamically, just need to add multiple components at once.

**Reply by allenhsu-17live** [reply]
Hi , @jnmaloney thx for your reply , but i read about samples/gltf_viewer.cpp
but I can't find out any about dynamically component for Skeletal, because I need..

**Reply by jnmaloney** [reply]
I second the need for dynamic animators.

@allenhsu-17live Are your components rigged already?
